CMakeLists.txt

```diff
@@ -57,18 +57,19 @@ option(ENABLE_ADJUSTABLEHUD_MOD "Install the build‑in “AdjustableHud mod”
 
 if(ENABLE_TFE)
 	add_executable(tfe)
-	set_target_properties(tfe PROPERTIES OUTPUT_NAME "theforceengine")
+	set_target_properties(tfe PROPERTIES OUTPUT_NAME "theforceengine_target") # rename the binary to "theforceengine_target" to avoid conflicts with the directory name on mac os x
 
 	if(UNIX)
 		find_package(PkgConfig REQUIRED)
-		find_package(SDL2 2.0.20 REQUIRED)
-		pkg_check_modules(SDL2_IMAGE REQUIRED SDL2_image)
+		pkg_check_modules(SDL2 REQUIRED sdl2)
+    pkg_check_modules(SDL2_IMAGE REQUIRED SDL2_image)
+
 		target_include_directories(tfe PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
-		target_include_directories(tfe PRIVATE ${SDL2_INCLUDE_DIRS})
-		target_include_directories(tfe PRIVATE ${SDL2_IMAGE_INCLUDE_DIRS})
-		target_link_libraries(tfe PRIVATE SDL2::SDL2main SDL2::SDL2
-					${SDL2_IMAGE_LIBRARIES}
```

Comment:
Out of curiousity, why does this not work for you? Given a recent enough version of sdl2 and cmake, this should work on all platforms.

Reply:
In the original setup, the linker couldn't find `SDL2_image`. Honestly, I’m not sure why – it’s something I don’t understand well enough to explain. 

All I know is that it worked after explicitly adding: 
```cmake
target_link_directories(tfe PRIVATE ${SDL2_LIBRARY_DIRS} ${SDL2_IMAGE_LIBRARY_DIRS})
``` 

I noticed I changed a few more lines while troubleshooting this and haven't reverted them. I’ll put the originals back to keep the PR small.